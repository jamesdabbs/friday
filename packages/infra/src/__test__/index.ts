import { Testing } from 'cdk8s';

export function synth(construct: any, { name = 'construct', ...params }: Record<string, unknown> = {}) {
  const app = Testing.app();
  const chart = new construct(app, name, params);
  return Testing.synth(chart);
}

export function find(chart: any[], kind: string, name?: string) {
  return chart.find(value => {
    if (value.kind !== kind) { return false }
    if (name && value.metadata?.name !== name) { return false }

    return true
  })
}
