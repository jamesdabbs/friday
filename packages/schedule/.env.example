GOOGLE_PROJECT_ID=...
GOOGLE_CLIENT_ID=...
GOOGLE_CLIENT_SECRET=...
CALENDAR_ID=...

# This should support a full OAuth signin flow, but for now uses a static user token
GOOGLE_PERSONAL_ACCESS_TOKEN=...
GOOGLE_PERSONAL_REFRESH_TOKEN=...
GOOGLE_REDIRECT_URI=http://schedule.localhost/auth/google/callback

GOOGLE_AUTH_URI=https://accounts.google.com/o/oauth2/auth
GOOGLE_TOKEN_URI=https://oauth2.googleapis.com/token
GOOGLE_AUTH_PROVIDER_X509_CERT_URL=https://www.googleapis.com/oauth2/v1/certs
