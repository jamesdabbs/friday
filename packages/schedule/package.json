{
  "name": "schedule",
  "version": "0.0.1",
  "description": "",
  "author": "James Dabbs <james.dabbs@gmail.com>",
  "private": true,
  "license": "UNLICENSED",
  "scripts": {
    "prebuild": "rimraf dist",
    "build": "nest build",
    "start": "nest start",
    "start:dev": "nest-platform dev",
    "start:debug": "nest start --debug --watch",
    "start:prod": "node dist/main"
  },
  "dependencies": {
    "@jamesdabbs/nest-platform": "0.0.5",
    "@nestjs/common": "^7.6.15",
    "@nestjs/config": "^0.6.3",
    "@nestjs/core": "^7.6.15",
    "@nestjs/platform-express": "^7.6.15",
    "googleapis": "^74.1.0",
    "reflect-metadata": "^0.1.13",
    "rimraf": "^3.0.2",
    "rxjs": "^6.6.6"
  },
  "devDependencies": {
    "@nestjs/cli": "^7.6.0",
    "@nestjs/schematics": "^7.3.0",
    "@types/express": "^4.17.11",
    "@types/node": "^14.14.36",
    "ts-loader": "^8.0.18",
    "ts-node": "^9.1.1",
    "tsconfig-paths": "^3.9.0",
    "typescript": "^4.2.3"
  }
}
