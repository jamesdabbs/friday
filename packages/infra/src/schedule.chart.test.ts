import { ScheduleChart } from './schedule.chart';
import { synth } from './__test__';

describe(ScheduleChart, () => {
  describe('with defaults', () => {
    const chart = synth(ScheduleChart)

    it('matches the snapshot', () => {
      expect(chart).toMatchSnapshot()
    })
  })
});
