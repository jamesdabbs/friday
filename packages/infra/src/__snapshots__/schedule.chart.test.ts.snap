// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleChart with defaults matches the snapshot 1`] = `
Array [
  Object {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": Object {
      "name": "schedule",
    },
    "spec": Object {
      "replicas": 1,
      "selector": Object {
        "matchLabels": Object {
          "cdk8s.deployment": "construct-schedule.deployment-c84b1450",
        },
      },
      "template": Object {
        "metadata": Object {
          "labels": Object {
            "cdk8s.deployment": "construct-schedule.deployment-c84b1450",
          },
        },
        "spec": Object {
          "containers": Array [
            Object {
              "env": Array [
                Object {
                  "name": "GOOGLE_CLIENT_ID",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_CLIENT_ID",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_CLIENT_SECRET",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_CLIENT_SECRET",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_PERSONAL_REFRESH_TOKEN",
                  "valueFrom": Object {
                    "secretKeyRef": Object {
                      "key": "GOOGLE_PERSONAL_REFRESH_TOKEN",
                      "name": "schedule-secret",
                    },
                  },
                },
                Object {
                  "name": "GOOGLE_CLIENT_REDIRECT_URI",
                  "value": "http://localhost:3001/auth/google/callback",
                },
              ],
              "image": "schedule",
              "imagePullPolicy": "IfNotPresent",
              "name": "schedule",
              "ports": Array [],
              "volumeMounts": Array [],
            },
          ],
          "volumes": Array [],
        },
      },
    },
  },
  Object {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": Object {
      "name": "schedule",
    },
    "spec": Object {
      "externalIPs": Array [],
      "ports": Array [
        Object {
          "port": 80,
          "targetPort": 3000,
        },
      ],
      "selector": Object {
        "cdk8s.deployment": "construct-schedule.deployment-c84b1450",
      },
      "type": "ClusterIP",
    },
  },
  Object {
    "apiVersion": "networking.k8s.io/v1beta1",
    "kind": "Ingress",
    "metadata": Object {
      "name": "schedule.ingress.local",
    },
    "spec": Object {
      "rules": Array [
        Object {
          "host": "schedule.localhost",
          "http": Object {
            "paths": Array [
              Object {
                "backend": Object {
                  "serviceName": "schedule",
                  "servicePort": 80,
                },
              },
            ],
          },
        },
      ],
    },
  },
]
`;
